### PR TITLE
testing methods and app 140 selectize inputs

### PR DESCRIPTION
**Pull request by schloerke**

### CODE REVIEW

File: srcjs/jster.ts [hunk 1]
```diff
@@ -148,14 +155,26 @@ class Jster {
     );
   }
 
+  waitFor(ms) {
+    this.add((done) => {
+      setTimeout(done, ms);
+    });
+  }
+  waitForShiny() {
```
Comment (wch) [inline srcjs/jster.ts]: Might want to replace this with event handlers that listen for `shiny:idle` and `shiny:busy` events. https://shiny.rstudio.com/articles/js-events.html

File: srcjs/methods/radio.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { $ } from "../globals";
+
+function options(id: string) {
```
Comment (wch) [inline srcjs/methods/radio.ts]: Remove this.

File: srcjs/methods/radio.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    .filter(".selectize-control")
+    .find(".selectize-dropdown-content")
+    .children();
+}
+
+function clickOption(id: string, value: string) {
+  $(`#${id} input[value='${value}']`).click();
+}
+
+function currentChoice(id: string) {
```
Comment (wch) [inline srcjs/methods/radio.ts]: `Option` and `Choice` should be made consistent somehow.

File: srcjs/methods/selectize.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+function currentChoice(id: string) {
+  return $(`#${id}`)
+    .siblings()
+    .filter(".selectize-control")
+    .find(".selectize-input")
+    .text();
+}
+
+function values(id: string) {
```
Comment (wch) [inline srcjs/methods/selectize.ts]: Should have a comment saying server-side selectize provides the first 1000 choices.

File: srcjs/methods/shiny.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+function isShinyBusy(): boolean {
+  if (!$) {
+    return false;
+  }
+  return $("html")
+    .first()
+    .hasClass("shiny-busy");
+}
+
+function waitForShiny(callback, timeout = 23) {
```
Comment (schloerke) [inline srcjs/methods/shiny.ts]: Use `jst.add(Jster.shiny.wait);` instead

File: srcjs/jster.ts [hunk 6]
```diff
@@ -148,14 +155,26 @@ class Jster {
     );
   }
 
+  waitFor(ms) {
```
Comment (schloerke) [inline srcjs/jster.ts]: rename to `wait`

File: srcjs/methods/shiny.ts (hunk 5 above)
Comment (schloerke) [reply srcjs/methods/shiny.ts]: do not rename in exports. make the name consistent

File: srcjs/methods/shiny.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  wait();
+}
+
+function hasOverlay() {
+  return $("#shiny-disconnected-overlay").length > 0;
+}
+
+export { waitForShiny as wait, isShinyBusy as isBusy, hasOverlay };
```
Comment (wch) [inline srcjs/methods/shiny.ts]: Suggestion: have consistent names here to reduce confusion.

**Comment by schloerke** [comment]
I'll move them to another repo in another PR. 😄 

Merging for now.

**Pull request merged**